src/fsharp/fsi/fsi.fs

```diff
@@ -2358,6 +2358,9 @@ type FsiEvaluationSession (fsiConfig: FsiEvaluationSessionHostConfig, argv:strin
     let defaultFSharpBinariesDir = System.AppDomain.CurrentDomain.BaseDirectory
     let currentDirectory = Directory.GetCurrentDirectory()
 #endif
+    // When used as part of FCS we cannot assume the current process is fsi.exe
+    // So we try to fallback to the default compiler dir.
+    let defaultFSharpBinariesDir = Internal.Utilities.FSharpEnvironment.BinFolderOfDefaultFSharpCompiler(None).Value
```

Comment:
@dsyme Thanks for looking into that.

The idea of this change is to improve the out-of-the box experience when there is no `FSharp.Core.dll` or rather `.optdata`/`.sigdata` in the current directory. When we use an optional parameter this doesn't really change and we can use/recommend using `-I:` as well, right?

Would it we OK to check if the current dictionary contains those files and then use either the system or the current directory?